src/Core/Resolvers/CosmosQueryEngine.cs

```diff
@@ -122,19 +174,19 @@ public class CosmosQueryEngine : IQueryEngine
                             new JProperty(QueryBuilder.PAGINATION_FIELD_NAME, jarray));
 
                         // This extra deserialize/serialization will be removed after moving to Newtonsoft from System.Text.Json
-                        return new Tuple<JsonDocument, IMetadata>(JsonDocument.Parse(res.ToString()), null);
```

Comment:
Thanks for taking care of this. Removing additional serialization step helps improve perf too. You should probably delete the comment too.

Reply:
sure.